app/routes/home.tsx

```diff
@@ -1,78 +1,102 @@
-import "~/styles/home.css";
+import { ErrorResponse } from "~/models/Error.model";
 
-interface Target {
-  value: string;
-}
+import { setAuthSession } from "~/util/auth/auth.server";
+import { createSupabaseServerClient } from "~/util/supabase/supabase.server";
+import { getSession, commitSession } from "~/util/supabase/supabase.server";
+import { signUp } from "~/api/signUp";
+import { signIn } from "~/api/signIn";
 
-interface Event {
-  target: Target;
-  preventDefault: VoidFunction;
-}
+type ActionData = {
+  _action: string;
+  error: ErrorResponse;
+};
 
-const HomePage: React.FC = () => {
-  // const { postLoginAndRedirect, postProject } = useContext(ApiContext);
-  const { supabase } = useContext(DatabaseContext);
-  const [newProjectName, setNewProjectName] = useState("");
-  const [newProjectPassword, setNewProjectPassword] = useState("");
-  const [loginProjectName, setLoginProjectName] = useState("");
-  const [loginProjectPassword, setLoginProjectPassword] = useState("");
-  const [loginErrorResponse, setLoginErrorResponse] = useState<ErrorResponse>(
-    {}
-  );
-  const [newProjectErrorResponse, setNewProjectErrorResponse] =
-    useState<ErrorResponse>({});
-
-  const handleLoginName = (event: Event) => {
-    setLoginProjectName(event.target.value);
-  };
-
-  const handleLoginPassword = (event: Event) => {
-    setLoginProjectPassword(event.target.value);
-  };
-
-  const handleLogin = (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    // postLoginAndRedirect(loginProjectName, loginProjectPassword).catch(
-    //   (error: ErrorResponse) => {
-    //     setLoginErrorResponse(error);
-    //   }
-    // );
-  };
-
-  const handleNewProjectName = (event: Event) => {
-    setNewProjectName(event.target.value);
-  };
-
-  const handleNewProjectPassword = (event: Event) => {
-    setNewProjectPassword(event.target.value);
-  };
-
-  const createProjectWithName = (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    const req = new Request("/.netlify/functions/create-new-project", {
-      method: "POST",
-      body: JSON.stringify({
-        name: newProjectName,
-        password: newProjectPassword,
-      }),
+import homeStyles from "~/styles/home.css";
+
+export let links: LinksFunction = () => {
+  return [{ rel: "stylesheet", href: homeStyles }];
+};
+
+export async function action({ request }: ActionFunctionArgs) {
+  let session = await getSession(request.headers.get("Cookie"));
+
+  let formData = await request.formData();
+  let _action = String(formData.get("_action"));
+  let email = String(formData.get("email"));
+  let password = String(formData.get("password"));
+
+  let supabaseClient = createSupabaseServerClient(request);
+
+  if (_action == "sign-up") {
+    let { accessToken, refreshToken, error } = await signUp(
+      email,
+      password,
+      supabaseClient
+    );
+
+    if (error) {
+      return { _action, error: error };
```

Comment:
the convention behind prefixing var names with `_` is to note them as private. Return statements should generally not include that. The preferred interface here is
```
return { action:_action, error}
```

Reply:
Understood, thanks for the reminder.